server/planning/feeding_services/onclusive_api_service.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from datetime import datetime
+import requests
+from superdesk.io.feeding_services.http_base_service import HTTPFeedingServiceBase
+from flask import current_app as app
+
+logger = logging.getLogger(__name__)
+
+
+class OnclusiveApiService(HTTPFeedingServiceBase):
+    """
+    Feeding Service class which can read events using HTTP
+    """
+
+    NAME = "Onclusive_api"
+    label = "Onclusive api feed"
+    service = "events"
+
+    fields = []
+
+    HTTP_AUTH = False
+    session = requests.Session()
+
+    current_date = datetime.now()
+
+    def _update(self, provider, update):
+        """
+        Fetch events from external API.
+
+        :param provider: Ingest Provider Details.
+        :type provider: dict
+        :param update: Any update that is required on provider.
+        :type update: dict
+        :return: a list of events which can be saved.
+        """
+        URL = app.config["ONCLUSIVE_URL"]
+        USERNAME = app.config["ONCLUSIVE_USERNAME"]
+        PASSWORD = app.config["ONCLUSIVE_PASSWORD"]
+        TOKEN = app.config["ONCLUSIVE_TOKEN"]
+        REF_TOKEN = app.config["ONCLUSIVE_REF_TOKEN"]
+
+        # authntication get token
+        if not TOKEN:
+            auth_url = URL + "api/v2/auth"
+            body = {"username": USERNAME, "password": PASSWORD}
+            resp = self.session.post(auth_url, body)
+
+            resp.raise_for_status()
+            if resp.status_code == 200 and resp.content:
+                data = resp.json()
+                if data.get("token") and data.get("refreshToken"):
+                    TOKEN = data["token"]
+                    REF_TOKEN = data["refreshToken"]
+
+        if TOKEN:
+            current_url = "{}/api/v2/events/date?date={}".format(URL, self.current_date.strftime("%Y%m%d"))
+            latest_url = "{}/api/v2/events/latest?date={}".format(URL, self.current_date.strftime("%Y%m%d"))
```

Comment:
I think we need to call current_url only during first ingest run and after that it should only call that latest url to get the changes. now you get everything for current date and then you get also new and update since that date which should be pretty much the same. might need also some pagination to fetch all events in the future

Reply:
maybe we should work with some upper limit to only fetch events in next 30 days or so and not everything, that db might be huge

Reply:
I see, Is there any way you remember to check for first ingest run?
Because I think for achieving this functionality we need to define a global variable something like count.

Reply:
it's stored on provider which you should be able to get in the service via `self._provider` https://github.com/superdesk/superdesk-core/blob/develop/superdesk/io/ingest_provider_model.py#L109